File: test/main/Models/Buildable/Material/LumberTest.java
package main.Models.Buildable.Material;

import main.Models.Measurement;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class LumberTest {
    @Test
    public void lumberShouldThrowWhenLengthIsTooLong() {
        IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class,
                () -> new Lumber(new Measurement(240, Measurement.Fraction.ONE_SIXTEENTH), Lumber.Dimension.TWO_BY_FOUR));
        String message = "A FactoryLength cannot be greater than 240\", was actually 240-1/16\"";
        assertEquals(message, thrown.getMessage());

        assertDoesNotThrow(() -> new Lumber(new Measurement(240), Lumber.Dimension.TWO_BY_FOUR));
    }

    @Test
    public void lumberShouldCompareEqual() {
        Lumber lumber1 = new Lumber(new Measurement(3), Lumber.Dimension.TWO_BY_FOUR);
        Lumber lumber2 = new Lumber(new Measurement(3), Lumber.Dimension.TWO_BY_FOUR);
        assertTrue(() -> lumber1.equals(lumber2));

        Lumber lumber3 = new Lumber(new Measurement(10), Lumber.Dimension.TWO_BY_FOUR);
        assertTrue(() -> lumber1.equals(lumber3));

        Lumber lumber4 = new Lumber(new Measurement(50), Lumber.Dimension.TWO_BY_FOUR);
        assertFalse(() -> lumber1.equals(lumber4));
    }

    @Test
    public void lumberShouldProduceUniqueHash() {
        int result = -1197299477;
        assertEquals(result, new Lumber(new Measurement(3), Lumber.Dimension.TWO_BY_FOUR).hashCode());

        int secondResult = -361291309;
        assertEquals(secondResult, new Lumber(new Measurement(92, Measurement.Fraction.FIVE_EIGHTH),
                Lumber.Dimension.TWO_BY_FOUR).hashCode());
    }

    @Test
    public void lumberShouldProduceNiceString() {
        String fourFootResult = "48\" 2x4";
        assertEquals(fourFootResult, new Lumber(new Measurement(40), Lumber.Dimension.TWO_BY_FOUR).toString());
    }
}

File: src/test/kotlin/models/buildable/Material/LumberTest.kt
package models.buildable.material

import models.Measurement
import models.Measurement.Fraction
import kotlin.test.*

internal class LumberTest {
    @Test
    fun lumberShouldThrowWhenLengthIsTooLong() {
        val thrown = assertFailsWith<IllegalArgumentException> {
            Lumber(Measurement(240, Fraction.ONE_SIXTEENTH), Lumber.Dimension.TWO_BY_FOUR)
        }
        val message = "A length cannot be greater than 240\", was 240-1/16\""
        assertEquals(message, thrown.message)
    }

    @Test
    fun lumberShouldCompareEqual() {
        val lumber1 = Lumber(Measurement(3), Lumber.Dimension.TWO_BY_FOUR)
        val lumber2 = Lumber(Measurement(3), Lumber.Dimension.TWO_BY_FOUR)
        assertTrue { lumber1 == lumber2 }
        val lumber3 = Lumber(Measurement(10), Lumber.Dimension.TWO_BY_FOUR)
        assertTrue { lumber1 == lumber3 }
        val lumber4 = Lumber(Measurement(50), Lumber.Dimension.TWO_BY_FOUR)
        assertFalse { lumber1 == lumber4 }
    }

    @Test
    fun lumberShouldProduceUniqueHash() {
        val result = 2142734874
        assertEquals(result, Lumber(Measurement(3), Lumber.Dimension.TWO_BY_FOUR).hashCode())
        val secondResult = -1041948622
        assertEquals(
            secondResult,
            Lumber(Measurement(92, Fraction.FIVE_EIGHTH),Lumber.Dimension.TWO_BY_FOUR,).hashCode(),
        )
    }

    @Test
    fun lumberShouldProduceNiceString() {
        val fourFootResult = "48\" 2x4"
        assertEquals(fourFootResult, Lumber(Measurement(40), Lumber.Dimension.TWO_BY_FOUR).toString())
    }
}